src/rdkafka_telemetry_decode.c

```diff
@@ -0,0 +1,536 @@
+                                                          'P',  'P', 'Y', 0};
+        static const size_t snappy_java_hdrlen         = 8 + 4 + 4;
+
+        /* snappy-java adds its own header (SnappyCodec)
+         * which is not compatible with the official Snappy
+         * implementation.
+         *   8: magic, 4: version, 4: compatible
+         * followed by any number of chunks:
+         *   4: length
+         * ...: snappy-compressed data. */
+        if (likely(inlen > snappy_java_hdrlen + 4 &&
+                   !memcmp(inbuf, snappy_java_magic, 8))) {
+                /* snappy-java framing */
+                char errstr[128];
+
+                inbuf = inbuf + snappy_java_hdrlen;
+                inlen -= snappy_java_hdrlen;
+                iov.iov_base = rd_kafka_snappy_java_uncompress(
+                    inbuf, inlen, &iov.iov_len, errstr, sizeof(errstr));
+
+                if (unlikely(!iov.iov_base)) {
+                        rd_rkb_dbg(
+                            rkb, MSG, "SNAPPY",
+                            "Snappy decompression for message failed: %s: "
+                            "ignoring message",
+                            errstr);
+                        return -1;  // Indicates decompression error
+                }
+
+
+        } else {
+                /* No framing */
+
+                /* Acquire uncompressed length */
+                if (unlikely(!rd_kafka_snappy_uncompressed_length(
+                        inbuf, inlen, &iov.iov_len))) {
+                        rd_rkb_dbg(
+                            rkb, MSG, "SNAPPY",
+                            "Failed to get length of Snappy compressed payload "
+                            "for message (%" PRIusz
+                            " bytes): "
+                            "ignoring message",
+                            inlen);
+                        return -1;  // Indicates decompression error
+                }
+
+                /* Allocate output buffer for uncompressed data */
+                iov.iov_base = rd_malloc(iov.iov_len);
+                if (unlikely(!iov.iov_base)) {
+                        rd_rkb_dbg(rkb, MSG, "SNAPPY",
+                                   "Failed to allocate Snappy decompress "
+                                   "buffer of size %" PRIusz
+                                   " for message (%" PRIusz
+                                   " bytes): %s: "
+                                   "ignoring message",
+                                   *outbuf_len, inlen, rd_strerror(errno));
+                        return -1;  // Indicates memory allocation error
+                }
+
+                /* Uncompress to outbuf */
+                if (unlikely((r = rd_kafka_snappy_uncompress(inbuf, inlen,
+                                                             iov.iov_base)))) {
+                        fprintf(stderr, "r: %d err %s\n", r,
+                                rd_strerror(errno));
```

Comment:
Remove this and other fprintf in this file

Reply:
Removed all fprintf